src/core/utils/helpers/general.ts

```diff
@@ -329,6 +329,20 @@ export const getPhysicalQueuedReservations = (list: ReservationType[]) => {
   );
 };
 
+export const loansOverdue = (loans: LoanType[]): boolean => {
+  let areAllLoansOverdue = true;
```

Comment:
Just a suggestion:
Why not use `.every()` here as you do in line 343? Then you can skip the `let`.
I did not know of the `every()`function. It is quite nifty :)